Frameworks/CoreText/DWriteWrapper_CTFont.mm

```diff
@@ -208,6 +208,173 @@ HRESULT _DWriteCreateFontFaceWithFontDescriptor(CTFontDescriptorRef fontDescript
     return E_INVALIDARG;
 }
 
+// Helper methods for removing fonts that do not match given attributes for CTFontDescriptorCreateMatchingFontDescriptors
+#pragma region Helper Filter Methods
+
+void __FilterFontsByTraits(std::vector<ComPtr<IDWriteFont>>& fonts, CFDictionaryRef traits, bool forceMatch) {
+    _DWriteFontProperties properties = __DWriteFontPropertiesFromTraits(traits);
+    std::vector<ComPtr<IDWriteFont>> matchingFonts;
+    matchingFonts.reserve(fonts.size());
+    for (auto& font : fonts) {
+        if (font->GetWeight() == properties.weight && font->GetStretch() == properties.stretch && font->GetStyle() == properties.style) {
+            matchingFonts.emplace_back(font);
+        }
+    }
+
+    if (!matchingFonts.empty() || forceMatch) {
+        fonts = std::move(matchingFonts);
+    }
+}
+
+HRESULT __FilterFontsByStyleName(std::vector<ComPtr<IDWriteFont>>& fonts, CFStringRef styleName, bool forceMatch) {
+    std::vector<ComPtr<IDWriteFont>> matchingFonts;
+    matchingFonts.reserve(fonts.size());
+    for (auto& font : fonts) {
+        ComPtr<IDWriteLocalizedStrings> dwriteFontName;
+        RETURN_IF_FAILED(font->GetFaceNames(&dwriteFontName));
+        CFStringRef name = _CFStringFromLocalizedString(dwriteFontName.Get());
+        if (CFStringCompare(name, styleName, kCFCompareCaseInsensitive) == kCFCompareEqualTo) {
+            matchingFonts.emplace_back(font);
+        }
+    }
+
+    if (!matchingFonts.empty() || forceMatch) {
+        fonts = std::move(matchingFonts);
+    }
+
+    return S_OK;
+}
+
+HRESULT __FilterFontsByFamilyName(std::vector<ComPtr<IDWriteFont>>& fonts, CFStringRef familyName, bool forceMatch) {
+    std::vector<ComPtr<IDWriteFont>> matchingFonts;
+    matchingFonts.reserve(fonts.size());
+    for (auto& font : fonts) {
+        // Need to create font family to get correct name
+        ComPtr<IDWriteFontFamily> family;
+        RETURN_IF_FAILED(font->GetFontFamily(&family));
+        ComPtr<IDWriteLocalizedStrings> dwriteFamilyName;
+        RETURN_IF_FAILED(family->GetFamilyNames(&dwriteFamilyName));
+        CFStringRef fontFamilyName = _CFStringFromLocalizedString(dwriteFamilyName.Get());
+        if (CFStringCompare(familyName, fontFamilyName, kCFCompareCaseInsensitive) == kCFCompareEqualTo) {
+            matchingFonts.emplace_back(font);
+        }
+    }
+
+    if (!matchingFonts.empty() || forceMatch) {
+        fonts = std::move(matchingFonts);
+    }
+
+    return S_OK;
+}
+
+HRESULT __FilterFontsByInformationalString(std::vector<ComPtr<IDWriteFont>>& fonts,
```

Comment:
I can strip them down until they're only loops w/ logic inside, but unfortunately the HRESULT returns aren't easily composable so I wouldn't be able to share any more without doing some crazy higher-order stuff that (though it would be pretty neat imo) almost certainly isn't worth it.